plaid/Cargo.toml

```diff
@@ -38,6 +38,8 @@ warp = { version = "0.3", features = ["tls"] }
 wasmer = { version = "4", default-features = false, features = ["cranelift"] }
 wasmer-middlewares = "4"
 jsonwebtoken = { version = "9.2" }
+tokio-tungstenite = { version = "0.23.1", features = ["native-tls"] }
+futures-util = "0.3.30"
```

Comment:
Tokio offers the [`split`](https://docs.rs/tokio/1.39.3/tokio/io/fn.split.html) method that would allow us to split the socket into a read half and a write half. 

Unfortunately, `WebSocketStream<tokio_tungstenite::MaybeTlsStream<tokio::net::TcpStream>>` does not satisfy the `AsyncRead` trait so we are unable to use tokio